examples/07_physics/main.rs

```diff
@@ -0,0 +1,276 @@
+//! Runs an nphysics integration demo
+extern crate amethyst;
+extern crate cgmath;
+extern crate rand;
+extern crate nalgebra;
+extern crate ncollide;
+extern crate nphysics3d;
+extern crate obj;
+
+use std::sync::{Mutex, Arc};
+use std::ops::{Deref, DerefMut};
+
+use amethyst::config::Element;
+use amethyst::context::Context;
+use amethyst::context::asset_manager::{Mesh, Texture};
+use amethyst::ecs::{Component, Join, Processor, RunArg, VecStorage, World};
+use amethyst::engine::{Application, State, Trans};
+use amethyst::processors::rendering::{Camera, Light, Projection, RenderingProcessor, Renderable};
+use amethyst::processors::transform::{Child, Init, LocalTransform, Transform, TransformProcessor};
+use amethyst::renderer::Light as RendererLight;
+use nalgebra::Vector3;
+use ncollide::shape::{Ball as PhysicsBall, Cuboid};
+use nphysics3d::object::{RigidBody, RigidBodyHandle};
+use nphysics3d::world as nphysics_world;
+
+
+struct PhysicsExample;
+
+// Wrap the nphysics3d `World` struct in our own anonymous struct
+// so that we can impl `Send`/`Sync` for it.
+struct PhysicsWorld(nphysics_world::World<f32>);
```

Comment:
Wondering if there's any interest in re-exporting this from within Amethyst, to avoid the `Send`/`Sync` and `Deref` hassle. Does it make sense to add it as a built-in `Processor`, to achieve something like `use amethyst::processors::physics::World;`? At any rate, probably something for another PR.